source/components/Autocomplete.js

```diff
@@ -192,24 +196,44 @@ class AutocompleteBase extends Component<AutocompleteProps, State> {
       !maxSelections ||
       (canMoreOptionsBeSelected && areFilteredOptionsAvailable)
     ) {
-      if (!selectedOption) return;
-      const option = selectedOption.trim();
-      const optionCanBeSelected =
-        (selectedOptions.indexOf(option) < 0 && !multipleSameSelections) ||
-        multipleSameSelections;
-
-      if (option && optionCanBeSelected && isOpen) {
-        const newSelectedOptions = _.concat(selectedOptions, option);
-        this.selectionChanged(newSelectedOptions, event);
-        this.setState({ selectedOptions: newSelectedOptions, isOpen: false });
+      if (!selectedOption || !selectedOption.length) return;
+      const option = _.isString(selectedOption) ? selectedOption.trim() : selectedOption;
+      let newSelectedOptions = [];
+      let skipValuesSelection = false;
+      if (option && Array.isArray(option)) {
+        newSelectedOptions = selectedOptions;
```

Comment:
I think here it would be easier to understand if you spread the `selectedOptions` into a new array like `[...selectedOptions]` … I was a bit confused when I read over this line + looked at the condition in the forEach afterward - because it seemed like these two arrays would be the same. Only then I saw that below you are assigning new arrays via `concat` 😉

Reply:
Fixed